File: easy/two_sum.py
class Solution:
    @staticmethod
    def two_sum(nums: list[int], target: int) -> list[int]:
        saved = dict()
        for index, num in enumerate(nums):
            needed = target - num
            needed_index = saved.get(needed)
            if needed_index is not None:
                return [needed_index, index]
            else:
                saved[num] = index
        return [0, 0]

if __name__ == '__main__':
    print(Solution.two_sum([3, 3], 6))

File: easy/two_sum.go
package main

import "fmt"

func main(){
	fmt.Println(twoSum([]int{2,7,11,15}, 9))
	fmt.Println(twoSum([]int{3,2,4}, 6))
}

func twoSum(nums []int, target int) []int {
    mapped := make(map[int]int)

    for i, v := range nums {
		needed := target - v
        ind, matched := mapped[needed]

		if matched && ind != i {
			return []int{ind, i}
		}
        mapped[v] = i
        
    }

    return []int{}
}